modules/validations/outdated_tags.js

```diff
@@ -93,63 +158,148 @@ export function validationOutdatedTags() {
-      if (newTags.name) {
-        for (let i = 0; i < nsiKeys.length; i++) {
-          const k = nsiKeys[i];
-          if (!newTags[k]) continue;
-
-          const center = entity.extent(graph).center();
-          const countryCode = countryCoder.iso1A2Code(center);
-          const match = _nsi.matcher.matchKVN(k, newTags[k], newTags.name, countryCode && countryCode.toLowerCase());
-          if (!match) continue;
-
-          // for now skip ambiguous matches (like Target~(USA) vs Target~(Australia))
-          if (match.d) continue;
-
-          const brand = _nsi.brands[match.kvnd];
-          if (brand && brand.tags['brand:wikidata'] &&
-            brand.tags['brand:wikidata'] !== entity.tags['not:brand:wikidata']) {
-            subtype = 'noncanonical_brand';
-
-            const keepTags = ['takeaway'].reduce((acc, k) => {
-              if (newTags[k]) {
-                acc[k] = newTags[k];
-              }
-              return acc;
-            }, {});
-
-            nsiKeys.forEach(k => delete newTags[k]);
-            Object.assign(newTags, brand.tags, keepTags);
-            break;
+        if (!loc) {     // collect location for this feature only once
+          loc = entity.extent(graph).center();
+        }
+        if (!names) {   // collect names for this feature only once
+          names = [];
+          let osmkeys = Object.keys(newTags);
+          for (let j = 0; j < osmkeys.length; j++) {
+            const k = osmkeys[j];
+            if (isNamelike(k, 'primary'))    names.unshift(newTags[k]);  // primary names first
+            if (isNamelike(k, 'alternate'))  names.push(newTags[k]);     // alternate names last
           }
+          if (foundQID) names.push(foundQID);  // matcher will recognize the QID as an alternate name too
+
+          names = utilArrayUniq(names).filter(Boolean);
+        }
+
+        // Try each namelike value
+        for (let n = 0; n < names.length; n++) {
+          match = _nsi.matcher.match(k, v, names[n], loc);   // Attempt to match an item in NSI
+          if (!match) continue;  // keep looking
+
+          // If we get here, there was a match..
+          // A match may contain multiple results, the first one is the best one for this location
+          // e.g. `['pfk-a54c14', 'kfc-1ff19c', 'kfc-658eea']`
+          const itemID = match[0].itemID;
+          const item = _nsi.ids.get(itemID);
+          const mainTag = item.mainTag;               // e.g. `brand:wikidata`
+          const itemQID = item.tags[mainTag];         // e.g. `brand:wikidata` qid
+          const notQID = newTags[`not:${mainTag}`];   // e.g. `not:brand:wikidata` qid
+
+          // Exceptions, throw out the match
+          if (
+            (!itemQID || itemQID === notQID) ||       // no `*:wikidata` or matched a `not:*:wikidata`
+            (newTags.office && !item.tags.office)     // feature may be a coprorate office for a brand? - #6416
```

Comment:
Typo:
```suggestion
            (newTags.office && !item.tags.office)     // feature may be a corporate office for a brand? - #6416
```

PRed into Codespell so would flag up eventually...

Reply:
fixed it in 3640e15fa, thanks for catching 👍

Reply:
Thanks!